test/e2e/framework/device_plugin_utils.go

```diff
@@ -0,0 +1,221 @@
+	ResourceName = "resource.name/1"
+)
+
+func NumberOfStubDevices(node *v1.Node) int64 {
+	val, ok := node.Status.Capacity[v1.ResourceName(ResourceName)]
+
+	if !ok {
+		return 0
+	}
+
+	return val.Value()
+}
+
+// StubDevicePlugin returns the stub Device Plugin pod for conformance testing
+func StubDevicePlugin(ns string) *v1.Pod {
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stub-device-plugin-" + string(uuid.NewUUID()),
+			Namespace: ns,
+		},
+
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyAlways,
+			Volumes: []v1.Volume{{
+				Name: "device-plugin-path",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: pluginapi.DevicePluginPath,
+					},
+				},
+			}, {
+				Name: "dummy-device-path",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/tmp/dummy-device",
+					},
+				},
+			}},
+			Containers: []v1.Container{
+				{
+					Name:  "stub-device-plugin",
+					Image: imageutils.GetE2EImage(imageutils.StubDevicePlugin),
+					// Args:  []string{"-resource_name", ResourceName},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "device-plugin-path",
+							MountPath: pluginapi.DevicePluginPath,
+						},
+						{
+							Name:      "dummy-device-path",
+							MountPath: "/tmp/dummy-device",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return p
+}
+
+// DummyDevice represents the dummy devices details with resource name.
+type DummyDevice struct {
+	ResourceName string             `json:"resourceName,omitempty"`
+	Devices      []pluginapi.Device `json:"devices,omitempty"`
+}
+
+// DummyDeviceList is a list of DummyDevice objects.
+type DummyDeviceList struct {
+	Items []DummyDevice `json:"items,omitempty"`
+}
+
+// CreateDummyDeviceFileFromConfig reads config file and parse it to create dummy device files on node
+// dummy device directory looks like this:
+// |-- tmp
+//     |-- dummy-device
+//         |-- resource.name-1  # hacky here, replace "-" with "/" to follow kubernetes resource naming convention
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: unhealthy)
+//         |-- resource.name-2
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: healthy)
+//             |-- device-3 (content: unhealthy)
+//         |-- resource.name-3
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: healthy)
+func CreateDummyDeviceFileFromConfig() error {
+	// clean up dummy device dir if needed
+	dir := path.Join(os.TempDir(), DummyDeviceDir)
+	err := os.RemoveAll(dir)
+	if err != nil {
+		return err
+	}
+
+	err = os.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	dummyDeviceList := readConfigFile()
+
+	for i, item := range dummyDeviceList.Items {
+		if i == 0 {
+			// TODO(cph): only get the first resource name now,
+			// we can add more test by extending this
+			ResourceName = item.ResourceName
```

Comment:
I will remove `ResourceName` const and get it from config file.
For now, one device plugin only support to register one resource, but `config file` maybe have more than one resources, so we pick up the first one here.

I have a new idea about this, maybe we can pass `ResourceName`/`ResourcePath` to stub-device-plugin container and it will read dummy device files from this `Path`.
Also we will deploy more than one stub-device-plugin in one test to simulate more resources registration.

WDYT?